extends.d.ts

```diff
@@ -0,0 +1,11 @@
+export {}; // Ensure this file is treated as a module to avoid global scope TS error
+
+declare global {
+  // support process.env.NODE_ENV === '...'
+  const process: {
+    env: {
+      NODE_ENV?: string;
+      [key: string]: string | undefined;
```

Comment:
Are we using any other env vars in our library code that are expected to exist on the user side? Otherwise I'd really lock this down to `NODE_ENV`.

Reply:
Yes: https://github.com/search?q=repo%3Amui%2Fmui-x+process.env.+path%3Apackages%2Fx-telemetry&type=code
However, we could have a specific file just for telemetry package, while others would use this one, WDYT?

Reply:
I was under the impression x-telemetry runs in Node.js, can't we just add `@types/node` in its tsconfig?

Reply:
It has installation/initialization phase, which runs on "node", but it also has a runtime, which sends requests on a website development build. 🤔

Reply:
Nevermind, I have mixed something up during various experiments, `vite/client` types cover the `@mui/x-telemetry` package needs.
I have removed the loose env variable typing. 😉